src/olympia/files/views.py

```diff
@@ -204,3 +206,20 @@ def serve(request, viewer, key):
         fobj, as_attachment=True, content_type=obj['mimetype'])
 
     return response
+
+
+def serve_file_upload(request, uuid):
+    upload = shortcuts.get_object_or_404(FileUpload, uuid=uuid)
+    access_token = request.GET.get('access_token')
+
+    if not access_token:
+        log.error('Denying access to %s, no token.' % upload.id)
+        raise PermissionDenied
+
+    if access_token != upload.access_token:
```

Comment:
If we want to be serious about it... Use `constant_time_compare` from `django.utils.crypto`.

Reply:
oh, yeah. That's a good call, I forgot about timing attack on string comparison..

Reply:
To be fair... to exploit this, you'd need to guess the uuid of your target (good luck!), find the token using some kind of timing attack, and do all that for the very limited amount of time the FileUpload still have an xpi attached... all this just to download an xpi :)